### PR TITLE
Added new file format param to skops snippet script

### PR DESCRIPTION
**Pull request by merveenoyan**
As discussed with @rajshah4 in Slack, it's better to state explicitly to load a model using pickle if it's a pickle file. Also cc @adrinjalali

### CODE REVIEW

File: js/src/lib/interfaces/Libraries.ts
Comment (merveenoyan) [reply js/src/lib/interfaces/Libraries.ts:230]: Back when I wrote this we decided instead of repetitively changing it we could simply set it for once. I think comment suffices, WDYT?

Comment (adrinjalali) [reply js/src/lib/interfaces/Libraries.ts:230]: Sure, but something's not working here, as in, when the extension is not `skops`, it shouldn't show this but it is showing it.

Comment (merveenoyan) [reply js/src/lib/interfaces/Libraries.ts:230]: I feel like it would be cool if we had a section in config.json that states which format the file is serialized with so Hub could use it to infer the snippet. WDYT

Comment (adrinjalali) [reply js/src/lib/interfaces/Libraries.ts:230]: that's what I suggested in the original PR you had I think, I don't mind that.

Comment (osanseviero) [reply js/src/lib/interfaces/Libraries.ts:230]: Any updates on this?

Comment (merveenoyan) [reply js/src/lib/interfaces/Libraries.ts:230]: @osanseviero there was a renaming around our serialization & loading functions (and a debate around it since planned renaming was causing problems as well) I will make more changes accordingly.

Comment (osanseviero) [reply js/src/lib/interfaces/Libraries.ts:230]: Awesome! Thanks a lot! :fire:

Comment (merveenoyan) [reply js/src/lib/interfaces/Libraries.ts:230]: Can we simply merge this with additional note for now?
We have a [prerequisite issue](https://github.com/skops-dev/skops/issues/228) to determine file format (now that model format is merged) and in model card, we decided for now we depend on file extension to add the code to get started. (so users can depend on that) 
Other option for now is to infer this snippet from file extension which was previously objected. (as far as I remember)

**Comment by osanseviero** [comment]
The current code snippet does not work I think. E.g. go to https://huggingface.co/Ramos-Ramos/emb-gam-dino-resnet 

The snippet is

```python
from skops.hub_utils import download
from skops.io import load

download("Ramos-Ramos/emb-gam-dino-resnet", "path_to_folder")
# make sure model file is in skops format
# if model is a pickle file, make sure it's from a source you trust
model = load("path_to_folder/model.pkl")
```

If you try the above, it will give an error

```
 TypeError: download() takes 2 positional arguments but 1 was given
```

The expectation is actually to use named arguments
```python
download(repo_id="Ramos-Ramos/emb-gam-dino-resnet", dst="emb-gam-dino-resnet")
```

But then

```python
model = load("path_to_folder/model.pkl")
```

Gives me an error

```
BadZipFile: File is not a zip file
```

What did work for me was

```
import pickle

with open("emb-gam-dino-resnet/model.pkl", "rb") as file: 
  logistic_regression = pickle.load(file)
```

Could we update the auto-snippet for it to work end to end?

File: js/src/lib/interfaces/Libraries.ts
Comment (osanseviero) [inline js/src/lib/interfaces/Libraries.ts:241]: Should we still have this warning but in the pickle snippet?

File: js/src/lib/interfaces/Libraries.ts
Comment (osanseviero) [inline js/src/lib/interfaces/Libraries.ts:239]: Please add the imports at the top, and also with `from skops.hub_utils import download` which is not in this one

File: js/src/lib/interfaces/Libraries.ts
Comment (merveenoyan) [reply js/src/lib/interfaces/Libraries.ts:241]: I just removed it, was a merge conflict

**Comment by merveenoyan** [comment]
@osanseviero please don't review until I can test it 😅 (had a problem in codespaces)

**Comment by merveenoyan** [comment]
1. model uploaded using skops with pickle
Ekran Resmi 2022-12-09 21 08 20
2. model uploaded using skops with skops as file format
Ekran Resmi 2022-12-09 21 09 24
3. vanilla sklearn model
Ekran Resmi 2022-12-09 21 10 07

**Comment by merveenoyan** [comment]
The work on this is done: https://github.com/skops-dev/skops/pull/242 (just made a mistake on changelog, fixed it) we can merge this one after that one.

File: js/src/lib/interfaces/Libraries.ts
Comment (osanseviero) [inline js/src/lib/interfaces/Libraries.ts:242]: Should this have a security warning as the one we had before? ("if model is a pickle file, make sure it's from a source you trust")

File: js/src/lib/interfaces/Libraries.ts
Comment (osanseviero) [inline js/src/lib/interfaces/Libraries.ts:253]: Should we bring the security warning back?

Comment (osanseviero) [inline js/src/lib/interfaces/Libraries.ts:253]: Please unindent the closing `)` in the vanilla sklearn code snippet

**Review by osanseviero — APPROVED** [review]
The results look nice! I think we might want to:

* Keep showing security message for joblib as we were doing before
* Looking at the file (https://github.com/huggingface/hub-docs/blob/02dc01dcab138b07081c6938964198f0129252a4/js/src/lib/interfaces/Libraries.ts#L236), there are some minor indentation/formatting issues in the sklearn function. Please try to be consistent (e.g. opening curly braces in same line - 237/238; double indent in lines 247 and 254)

File: js/src/lib/interfaces/Libraries.ts
Comment (merveenoyan) [reply js/src/lib/interfaces/Libraries.ts:242]: SGTM, but also wanted to ask @adrinjalali about it. WDYT?

Comment (adrinjalali) [reply js/src/lib/interfaces/Libraries.ts:242]: yeah that sounds good to me. We can also have a link to our docs: https://skops.readthedocs.io/en/stable/persistence.html

**Review by adrinjalali — APPROVED** [review]

File: js/src/lib/interfaces/Libraries.ts
Comment (merveenoyan) [reply js/src/lib/interfaces/Libraries.ts]: Done

Comment (merveenoyan) [reply js/src/lib/interfaces/Libraries.ts]: brought it back

**Comment by adrinjalali** [comment]
This looks good to me, but I'm more comfortable if @osanseviero has a final look before merging.

**Comment by merveenoyan** [comment]
I feel like I'm going crazy so I'll stop working today 😂 (for pickle model)

Ekran Resmi 2022-12-12 20 37 32
Ekran Resmi 2022-12-12 20 37 07

**BUT**
Ekran Resmi 2022-12-12 20 28 33

**Comment by osanseviero** [comment]
In your code snippet you do

```
const skopssaveFormat = model.config?.sklearn?.model_format;
```

But there is no `model_format`, there is `fileformat`. Hence, you don't enter the condition of line 237 (format == `pickle`), and you end up being in the `else`, which is what I see in the screenshot. Maybe this is a good signal of not using if/else here and rather if/else if/else, and in the else show some other snippet with a comment saying it's not a supported format.

**Comment by merveenoyan** [comment]
@osanseviero screenshot has the one I'm using for build and it doesn't say 'model_format', it does 'fileformat', hence it gets into the if with 'pickle' (see what's logged)

**Comment by merveenoyan** [comment]
I guess I was using a wrong version on moon-landing and everything was working fine. @osanseviero 😂

Ekran Resmi 2022-12-13 15 50 35
Ekran Resmi 2022-12-13 15 48 08
Ekran Resmi 2022-12-13 15 51 30
We can merge after this one is merged: https://github.com/skops-dev/skops/pull/242

**Comment by merveenoyan** [comment]
@osanseviero updated variable name in accordance to moon-landing

**Comment by merveenoyan** [comment]
@osanseviero can we merge this since one in moon-landing is merged?

**Review by osanseviero — APPROVED** [review]
Feel free to merge :)

**Pull request merged**
